cirq/circuits/circuit.py

```diff
@@ -2200,6 +2201,31 @@ def with_noise(self, noise: 'cirq.NOISE_MODEL_LIKE') -> 'cirq.Circuit':
             c_noisy += Circuit(op_tree)
         return c_noisy
 
+    def decompose(self, add_empty_moments=True) -> Iterable['cirq.Circuit']:
+        """Decomposes circuit into a set of circuits, if it's possible (i.e. if
+        the circuit qubits can be divided into two or more groups of qubits
+        such that there are no entangling gates between them).
+
+        If this is not possible, will return the set consisting of the single
+        circuit (this one).
+        """
+        uf = networkx.utils.UnionFind(self.all_qubits())
+        for op in self.all_operations():
+            if len(op.qubits) > 1:
+                uf.union(*op.qubits)
+        qubit_sets = sorted(uf.to_sets(), key=lambda x: min(x))
+        if len(qubit_sets) == 1:
+            return (self,)
+        else:
```

Comment:
style nit: since you're `return`ing above, drop the `else` and de-dent the rest of the function.